File: nbdserver/clients/volumecontroller/VolumeInformation.go
package volumecontroller

import (
	"gopkg.in/validator.v2"
)

type VolumeInformation struct {
	Blocksize      int    `json:"blocksize" validate:"nonzero"`
	Deduped        bool   `json:"deduped" validate:"nonzero"`
	Driver         string `json:"driver,omitempty"`
	Id             string `json:"id" validate:"nonzero"`
	ReadOnly       bool   `json:"readOnly,omitempty"`
	Size           int    `json:"size" validate:"nonzero"`
	Storagecluster string `json:"storagecluster" validate:"nonzero"`
}

func (s VolumeInformation) Validate() error {

	return validator.Validate(s)
}

File: volumecontroller/pythonimp/VolumeInformation.py

from flask_wtf import Form
from wtforms.validators import DataRequired, Length, Regexp, NumberRange, required
from wtforms import TextField, FormField, IntegerField, FloatField, FileField, BooleanField, DateField, FieldList
from input_validators import multiple_of



class VolumeInformation(Form):
    
    blocksize = IntegerField(validators=[DataRequired(message="")])
    deduped = BooleanField(validators=[DataRequired(message="")])
    driver = TextField(validators=[])
    id = TextField(validators=[DataRequired(message="")])
    readOnly = BooleanField(validators=[])
    size = IntegerField(validators=[DataRequired(message="")])
    storagecluster = TextField(validators=[DataRequired(message="")])
